### PR TITLE
4493: User Case mapping

### PR DESCRIPTION
**Pull request by sutt0n**
- added new fields to `[ pk: user|..., sk: case|...]` item:
  - caseCaption
  - leadCaseId
  - status
  - docketNumber
  - docketNumberWithSuffix
  - caseId
  - createdAt

### CODE REVIEW

File: shared/src/business/entities/UserCase.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+UserCase.validationName = 'UserCase';
+
+/**
+ * UserCase Entity
+ * represents a user-to-case mapping record
+ *
+ * @param {object} rawUserCase the raw user-case data
+ * @constructor
+ */
+function UserCase(rawUserCase) {
```
Comment (rachaelparris) [inline shared/src/business/entities/UserCase.js:16]: Does this entity need validation?

File: shared/src/persistence/dynamo/cases/createCase.js [hunk 2]
```diff
@@ -65,10 +65,18 @@ exports.createCase = async ({ applicationContext, caseToCreate }) => {
         applicationContext,
       }),
     ),
+    // case:user mapping record (for dashboard)
     client.put({
```
Comment (rachaelparris) [inline shared/src/persistence/dynamo/cases/createCase.js]: I think you'll need to also update this record if any of these fields on the case are updated.

File: shared/src/persistence/dynamo/cases/createCase.js [hunk 3]
```diff
@@ -65,10 +65,18 @@ exports.createCase = async ({ applicationContext, caseToCreate }) => {
...
       Item: {
+        caseCaption: caseToCreate.caseCaption,
```
Comment (rachaelparris) [inline shared/src/persistence/dynamo/cases/createCase.js]: Shouldn't the `associateUserWithCase` persistence method be where this record is added?

**Review by kkoskelin — APPROVED** [review]
What she said. ^

File: shared/src/persistence/dynamo/cases/createCase.js (hunk 3 above)
Comment (sutt0n) [reply shared/src/persistence/dynamo/cases/createCase.js]: @rachaelparris yes, but this maps it to the user that created the case

File: shared/src/business/entities/UserCase.js (hunk 1 above)
Comment (sutt0n) [reply shared/src/business/entities/UserCase.js:16]: it was mainly used to whitelist fields from the Case entity; however, i can copy validation over

File: shared/src/persistence/dynamo/cases/createCase.js (hunk 3 above)
Comment (rachaelparris) [reply shared/src/persistence/dynamo/cases/createCase.js]: Could you call this as a separate persistence method in `createCaseInteractor`? We won't have a user to map in `createCaseFromPaperInteractor`, which also calls the `createCase` persistence method.

File: shared/src/persistence/dynamo/cases/updateCase.js [hunk 4]
```diff
@@ -234,8 +234,45 @@ exports.updateCase = async ({ applicationContext, caseToUpdate }) => {
     }
   }
 
+  // update user-case mappings
+  if (
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/cases/updateCase.js:238]: 👍

**Review by codyseibert — APPROVED** [review]

File: shared/src/persistence/dynamo/cases/createCase.js (hunk 3 above)
Comment (sutt0n) [reply shared/src/persistence/dynamo/cases/createCase.js]: the internal user is added to the mapping, but i see what you're saying. i'll do that, thanks!

File: shared/src/business/useCases/getConsolidatedCasesByUserInteractor.test.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ describe('getConsolidatedCasesByUserInteractor', () => {
     );
     applicationContext
       .getPersistenceGateway()
-      .getCasesByUser.mockImplementation(({ userId }) => {
```
Comment (rachaelparris) [inline shared/src/business/useCases/getConsolidatedCasesByUserInteractor.test.js:17]: Will `getCasesByUser` now become obsolete? Or does it have other uses? If it won't be needed, could you delete it? (in another PR is fine)

Comment (sutt0n) [reply shared/src/business/useCases/getConsolidatedCasesByUserInteractor.test.js:17]: it has other uses.

**Pull request merged**